pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -0,0 +1,343 @@
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get virtualmachine instance for the VM with name: %q. Error: %+v", instance.Spec.VMName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Found virtualMachine instance for VM with name: %q: +%v", instance.Spec.VMName, vm)
+
+		// Set ownerRef on CnsFileAccessConfig instance (in-memory) to VM instance
+		setInstanceOwnerRef(instance, instance.Spec.VMName, vm.UID)
+
+		cnsFinalizerExists := false
+		// Check if finalizer already exists.
+		for _, finalizer := range instance.Finalizers {
+			if finalizer == cnsoperatortypes.CNSFinalizer {
+				cnsFinalizerExists = true
+				break
+			}
+		}
+		// Update finalizer and ownerRef on CnsFileAccessConfig instance
+		if !cnsFinalizerExists {
+			// Add finalizer.
+			instance.Finalizers = append(instance.Finalizers, cnsoperatortypes.CNSFinalizer)
+			err = updateCnsFileAccessConfig(ctx, r.client, instance)
+			if err != nil {
+				msg := fmt.Sprintf("failed to update CnsFileAccessConfig instance: %q on namespace: %q. Error: %+v",
+					instance.Name, instance.Namespace, err)
+				recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
+				return reconcile.Result{RequeueAfter: timeout}, nil
+			}
+		}
+
+		volumeID, err := util.GetVolumeID(ctx, r.client, instance.Spec.PvcName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get volumeID from pvcName: %q. Error: %+v", instance.Spec.PvcName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// Query volume
+		log.Infof("Querying volume: %s for CnsFileAccessConfig request with name: %q on namespace: %q",
+			volumeID, instance.Name, instance.Namespace)
+		volume, err := common.QueryVolumeByID(ctx, r.volumeManager, volumeID)
+		if err != nil {
+			if err.Error() == common.ErrNotFound.Error() {
+				msg := fmt.Sprintf("CNS Volume: %s not found", volumeID)
+				log.Error(msg)
+				setInstanceError(ctx, r, instance, msg)
+				return reconcile.Result{RequeueAfter: timeout}, nil
+			}
+			log.Errorf("Failed to query CNS volume: %s with error: %+v", volumeID, err)
+			setInstanceError(ctx, r, instance, "Unable to find the volume in CNS")
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		vSANFileBackingDetails := volume.BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails)
+		accessPoints := make(map[string]string)
+		for _, kv := range vSANFileBackingDetails.AccessPoints {
+			accessPoints[kv.Key] = kv.Value
+		}
+		if len(accessPoints) == 0 {
+			log.Errorf("No access points found for volume: %q", volumeID)
+			setInstanceError(ctx, r, instance, "No access points found.")
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// Read the CnsFileAccessConfig instance again because the instance is already modified
+		err = r.client.Get(ctx, request.NamespacedName, instance)
```

Comment:
What about in other places between here and update on line 198? There are cases where the error is set on the object, so isnt there a need to `get()` before that too?

Reply:
Not needed, since other places are only changing the status(setInstanceError)

Reply:
`client.Update` modifies the input instance according to the documentation:

```
	// Update updates the given obj in the Kubernetes cluster. obj must be a
	// struct pointer so that obj can be updated with the content returned by the Server.
```

In that case, this `Get()` is redundant right?